docs: Restructure quick-start section of docs to separate clp-json from clp-text (#968)
y-scope/clp: docs/src/user-guide/quick-start/overview.md

Reviewer: Let's rename this to `index.md`. In web-programming `index.<ext>` is the entry point for a website, so following that convention here will make it more obvious for future developers.

# Overview

> This page will guide you through CLP's system requirements and help you choose which flavour of CLP will be best for you.
> 
> If you already have CLP set up, click your flavour of CLP below to learn how to use it.

Reviewer: ```suggestion
This guide describes:

- [CLP's system requirements](#system-requirements);
- [how to choose a CLP flavor](#choosing-a-flavour); and
- [how to use CLP](#using-clp).
```

* I think this is faster to read and navigate.
* Let's add a new section at the bottom of the page called "Using CLP", and then move the cards you have below into that section.

::::{grid} 1 1 2 2
:gutter: 2

:::{grid-item-card}
:link: clp-json
CLP for JSON logs
^^^
Learn about compressing and searching JSON logs.
:::

:::{grid-item-card}
:link: clp-text
CLP for unstructured text logs
^^^^^^^^^^^^
Learn about compressing and searching unstructured text logs.
:::
::::

---

## System Requirements

> To run properly on your system, CLP requires a few other programs. Before you set up CLP, ensure that you have the following installed.

Reviewer: ```suggestion
To run a CLP release, you'll need:

* [Docker](#docker)
* [Python](#python)
```

### Docker

> CLP uses Docker to deploy its different components. You can check whether Docker is already installed on your system by running the command

Reviewer: ```suggestion
To check whether Docker is installed on your system, run:
```

The additional information is useful, but probably too much for a quick-start guide.

```bash
docker version
```

> If Docker is not on your system, follow the instructions [here][Docker] to install it.

Reviewer: ```suggestion
If Docker isn't installed, follow these [instructions][Docker] to install it.
```

> If you're not running as root, ensure Docker can be run [without superuser privileges][docker-non-root].
> 
> If you're using Docker Desktop, ensure version 4.34 or higher is installed, and [host networking is enabled][docker-desktop-host-networking].

Reviewer: ```suggestion
NOTE:

* If you're not running as root, ensure Docker can be run [without superuser privileges][docker-non-root].
* If you're using Docker Desktop, ensure version 4.34 or higher is installed, and [host networking is enabled][docker-desktop-host-networking].
```

* I find lists are easier to read rather than independent lines of text.
* I'm suggesting just a plain "NOTE:" rather than a `{note}` admonition since I think readers have a tendency to skip the latter when they're rushing.

### Python

> CLP uses Python to interpret the scripts that coordinate how it runs. Specifically, CLP needs Python version 3.8 or higher. Many Linux distributions come with Python pre-installed; to confirm that it's on your system, run the command

Reviewer: ```suggestion
To check whether Python is installed on your system, run:
```

```bash
python3 --version
```

> If Python isn't on your system, or if the version isn't high enough, install or upgrade it.

Reviewer: ```suggestion
CLP requires Python 3.8 or higher. If Python isn't installed, or if the version isn't high enough, install or upgrade it by following the instructions for your OS.
```

:::{note}
If you're planning to deploy CLP on multiple nodes/systems, there are a few other system requirements; check out the [multi-node deployment](../guides-multi-node) page for more details.
:::

> 

Author: no, we changed that --- I have confirmed that the link works correctly.

> 

Reviewer: Organizationally, this is a bit awkward to place here (e.g., a reader might think it's part of the Python section). I think let's just delete it and the user can figure it out when they read the multi-node guide.

---

## Choosing a flavour

There are two flavours of CLP:

* **[clp-json](#clp-json)** for compressing and searching **JSON** logs.
* **[clp-text](#clp-text)** for compressing and searching **unstructured text** logs.

:::{note}
Both flavours contain the same binaries but are configured with different values for the
`package.storage_engine` key in the package's config file (`etc/clp-config.yml`).
:::

### clp-json

The JSON flavour of CLP is appropriate for JSON logs, where each log event is an independent JSON
object. For example:

```json lines
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "ctx": "conn11",
  "msg": "Waiting for write concern."
}
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "msg": "Set last op to system time"
}
```

The log file above contains two log events represented by two JSON objects printed one after the
other. Whitespace is ignored, so the log events could also appear with no newlines and indentation.

> If your logs are JSON logs, download and extract the JSON flavour tarball from the [Releases][clp-releases] page on GitHub, and proceed to the clp-json portal below.
> 
> ::::{grid} 1 1 1 1
> :gutter: 2
> 
> :::{grid-item-card}
> :link: clp-json
> CLP for JSON logs
> ^^^
> Learn how to start up clp-json, and begin compressing and searching JSON logs.
> :::
> ::::

Reviewer: ```suggestion
If you're using JSON logs, download and extract the `clp-json` release from the [Releases][clp-releases] page, then process to the [clp-json](./clp-json.md) quick-start guide.
```

I guess you included a card here to make it hard to miss, but I think if we add a "Using CLP" section at the bottom, that will address any wandering readers' needs.

Reviewer: Same comment for the clp-text section.

> 

Author: We'll include that in a video we're making
